### PR TITLE
Use ANSIBLE_BASE_JWT_KEY to make Galaxy aware of Envoy

### PR DESCRIPTION
**Pull request by bmclaughlin**
Specifically in the case of downloading collections. 

Before:
```
$ ansible-galaxy collection download -p . awx.awx -s https://localhost/api/hub/content/community/ --api-key=<INSERT_KEY_HERE> --ignore-certs --no-cache
Process download dependency map
Starting collection download process to '/home/bmclaugh/Downloads/_tmp'
Downloading http://localhost:5001/api/hub/v3/plugin/ansible/content/community/collections/artifacts/awx-awx-24.0.0.tar.gz to /home/bmclaugh/.ansible/tmp/ansible-local-3030078ycyczlke/tmpzw38d1oq/awx-awx-24.0.0-if08m8a7
Downloading collection 'awx.awx:24.0.0' to '/home/bmclaugh/Downloads/'
ERROR! Failed to download collection tar from 'cmd_arg' due to the following unforeseen error: HTTP Error 404: Not Found. HTTP Error 404: Not Found
```

After:
```
$ ansible-galaxy collection download -p . awx.awx -s https://localhost/api/hub/content/community/ --api-key=<INSERT_KEY_HERE> --ignore-certs --no-cache
Process download dependency map
Starting collection download process to '/home/bmclaugh/Downloads'
Downloading https://localhost/api/hub/v3/plugin/ansible/content/community/collections/artifacts/awx-awx-24.0.0.tar.gz to /home/bmclaugh/.ansible/tmp/ansible-local-2970413jpwpnnso/tmpbpkvb8my/awx-awx-24.0.0-nb1v7e1z
Downloading collection 'awx.awx:24.0.0' to '/home/bmclaugh/Downloads'
Collection 'awx.awx:24.0.0' was downloaded successfully
Writing requirements.yml file of downloaded collections to '/home/bmclaugh/Downloads/requirements.yml'
```

AAP-20998
No-Issue

### CODE REVIEW

File: galaxy_ng/app/dynaconf_hooks.py [hunk 1]
```diff
@@ -578,6 +580,27 @@ def configure_legacy_roles(settings: Dynaconf) -> Dict[str, Any]:
...
 
 
+def configure_aap_proxy(settings: Dynaconf) -> Dict[str, Any]:
+    ANSIBLE_BASE_JWT_KEY = settings.get("ANSIBLE_BASE_JWT_KEY")
+    if ANSIBLE_BASE_JWT_KEY:
+        data = {
+            "ANSIBLE_API_HOSTNAME": settings.get("ANSIBLE_API_HOSTNAME"),
+            "ANSIBLE_CONTENT_HOSTNAME": settings.get("ANSIBLE_CONTENT_HOSTNAME"),
+        }
+        gw_url = urlparse(ANSIBLE_BASE_JWT_KEY)
```
[the remark this answers is not in the document]
Comment (bmclaughlin) [reply galaxy_ng/app/dynaconf_hooks.py:590]: David commented on the jira issue (here's the excerpt) `ANSIBLE_BASE_JWT_KEY which can either be a URL or a key file`.  If `ANSIBLE_BASE_JWT_KEY` is a key the `scheme` and `hostname` will not be populated in the following check and nothing should change.

[the remark this answers is not in the document]
Comment (rochacbruno) [reply galaxy_ng/app/dynaconf_hooks.py:590]: So I think it is good to add a comment so the future us will know about it when dealing with this piece of code :) 

```python
# The following variable is either a URL or a key file path.
ANSIBLE_BASE_JWT_KEY = settings.get("ANSIBLE_BASE_JWT_KEY") 
```

**Review by jctanner — APPROVED** [review]

**Pull request merged**